openpathsampling/collectivevariables/plumed_wrapper.py

```diff
@@ -0,0 +1,326 @@
+            'components': self.components,
+            'kwargs': self.kwargs,
+            'cv_requires_lists': self.cv_requires_lists,
+            'cv_wrap_numpy_array': self.cv_wrap_numpy_array,
+            'cv_scalarize_numpy_singletons': self.cv_scalarize_numpy_singletons
+        }
+
+
+class PLUMEDInterface(StorableNamedObject):
+
+    """Interfaces the Cython PLUMED wrapper located at `/path/to/plumed2/python`
+    and allows to set and get non-`PLUMEDCV` commands (i.e., non-outputting).
+    This includes groups of atoms, centers of mass, include files, etc.
+    Requires PLUMED development version (see https://github.com/plumed/plumed2)
+    and sourcing `/path/to/plumed2/sourceme.sh`
+
+    References
+    ----------
+
+    .. [1] G.A. Tribello, M. Bonomi, D. Branduardi, C. Camilloni, G. Bussi,
+    PLUMED2: New feathers for an old bird, Comp. Phys. Comm. 185, 604 (2014);
+    https://doi.org/10.1016/j.cpc.2013.09.018
+
+    Examples
+    --------
+    >>> # To group the atoms [7,9,15,17] corresponding to the dihedral psi
+    >>> # in Ala dipeptide:
+    >>> from openpathsampling import PLUMEDCV, PLUMEDInterface
+    >>> plmd = PLUMEDInterface(top)
+    >>> # top is an `openpathsampling.engines.openmm.topology.MDTrajTopology`
+    >>> plmd.set("group","GROUP ATOMS=7,9,15,17")
+    >>> psi_plumed = PLUMEDCV("psi",plmd,"TORSION ATOMS=group")
+    >>> print psi_plumed(traj)  # returns psi values for the trajectory
+    >>> pld.get()  # returns (('group', 'GROUP ATOMS=7,9,15,17'))
+    """
+
+    def __init__(self,
+                 topology,
+                 pathtoplumed="",
+                 timestep=1.,
+                 kbt=1.,
+                 molinfo="",
+                 logfile="plumed.log"):
+        """
+        Parameters
+        ----------
+        topology : :obj:`openpathsampling.engines.openmm.MDTopology`
+        pathtoplumed : string
+            path to the PLUMED installation
+        timestep : double
+            Time step size of the simulation in PLUMED default units (ps).
+        kbt : double
+            :math:`$k_BT$` in PLUMED default units (kJ/mol).
+        molinfo : string
+            A PDB file containing information about the molecule. (see
+            https://plumed.github.io/doc-v2.4/user-doc/html/_m_o_l_i_n_f_o.html).
+        logfile : string
+            Name of the PLUMED log file.
+        """
+
+        self.interface = plumed.Plumed()  # 8 is default size of real
+        self.pathtoplumed = pathtoplumed
+        self.topology = topology
+        self.timestep = timestep
+        self.kbt = kbt
+        self.molinfo = molinfo
+        self.logfile = logfile
+        self._commandlist = []
+        self._init_plumed()
+
+    def cmd(self, *args, **kwargs):
+        self.interface.cmd(*args, **kwargs)
+
+    def _init_plumed(self):
+        if self.pathtoplumed != "":  # pragma: no cover
+            os.system("source " + self.pathtoplumed + "/sourceme.sh")
```

Comment:
1. Use `os.path.join(self.pathtoplumed, 'sourceme.sh')` to ensure that this will work if we ever get Windows support going.

2. This is a security risk ([command injection](https://www.kevinlondon.com/2015/07/26/dangerous-python-functions.html)). I think you can get the same effect with `subprocess.call`, with `shell=False`. If you need `shell=True`, you should add a safety check -- perhaps ensure that `os.path.join(self.pathtoplumed, 'sourceme.sh')` is a file, with `os.path.exists`.